core/src/main/java/org/apache/iceberg/PartitionStatsUtil.java

```diff
@@ -40,27 +44,48 @@ public class PartitionStatsUtil {
   private PartitionStatsUtil() {}
 
   /**
-   * Computes the partition stats for the given snapshot of the table.
+   * Fully computes the partition stats for the given snapshot of the table.
    *
    * @param table the table for which partition stats to be computed.
    * @param snapshot the snapshot for which partition stats is computed.
    * @return the collection of {@link PartitionStats}
    */
   public static Collection<PartitionStats> computeStats(Table table, Snapshot snapshot) {
-    Preconditions.checkArgument(table != null, "table cannot be null");
-    Preconditions.checkArgument(Partitioning.isPartitioned(table), "table must be partitioned");
-    Preconditions.checkArgument(snapshot != null, "snapshot cannot be null");
+    Preconditions.checkArgument(table != null, "Table cannot be null");
```

Comment:
FYI, in other places we typically use `Invalid table: null`. The same for snapshot: `Invalid snapshot: null`

Reply:
I see only in `TableUitl` the message you mentioned, in Flink, I saw some places it is similar to what is here. 

Since it is not standardized and we already shipped with that error message. I don't think we need to correct it.

Reply:
updated to `Invalid snapshot: null `and `Invalid table: null` syntax